src/shared/generator.c

```diff
@@ -694,6 +695,91 @@ int generator_hook_up_pcrfs(
         return generator_add_symlink_full(dir, where_unit, "wants", pcrfs_unit_path, instance);
 }
 
+int generator_hook_up_quotacheck(
+                const char *dir,
+                const char *what,
+                const char *where,
+                const char *target,
+                const char *fstype) {
+
+        const char *quotacheck_unit, *quotacheck_unit_path;
+        _cleanup_free_ char *where_unit = NULL, *instance = NULL;
+        int r;
+
+        assert(dir);
+        assert(where);
+
+        if (!isempty(fstype) && !streq(fstype, "auto")) {
+                if (!fstype_needs_quota(fstype))
+                        return log_warning_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Quota was requested for %s, but not supported, ignoring: %s", what, fstype);
+        } else
+                return log_warning_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Couldn't determine filesystem type for %s, Quota cannot be activated", what);
```

Comment:
- Please add period. `... activated.`
- The lines for logging are slightly long, please break lines.
- I'd like to invert the first condition, then the indentation can be reduced:
```c
if (isempty(fstype) || streq(fstype, "auto"))
        return log_warning_errno(...);
else if (!fstype_needs_quota(fstype))
        return log_warning_errno(...);
```